plugin.rb

```diff
@@ -94,7 +97,8 @@ class Engine < ::Rails::Engine
 
   add_to_serializer(:post, :current_user_reaction) do
     return nil unless scope.user.present?
-    object.reactions.includes([:reaction_users]).each do |reaction|
```

Comment:
This was converting a ActiveRecord collection proxy back into an association resulting a query being triggered when the associations are being accessed.